### PR TITLE
test: several changes for TEST-74-AUX-UTILS

### PR DESCRIPTION
**Pull request by yuwata**
For #32680.

### CODE REVIEW

**Comment by bluca** [comment]
```
22:17:49 [   29.668251] TEST-74-AUX-UTILS.sh[2173]: + timeout 60 bash -c 'while [[ -n $\(systemctl list-units --all --no-legend tmp-tmp.EyRYy4GEzu-mnt.mount\) ]]; do sleep 1; done'
22:17:49 [   29.674534] TEST-74-AUX-UTILS.sh[2200]: bash: -c: line 1: syntax error in conditional expression
22:17:49 [   29.674534] TEST-74-AUX-UTILS.sh[2200]: bash: -c: line 1: syntax error near `list-units'
22:17:49 [   29.674534] TEST-74-AUX-UTILS.sh[2200]: bash: -c: line 1: `while [[ -n $\(systemctl list-units --all --no-legend tmp-tmp.EyRYy4GEzu-mnt.mount\) ]]; do sleep 1; done'
```

**Comment by bluca** [comment]
`[  181.977183] TEST-74-AUX-UTILS.sh[428]: Subtest /usr/lib/systemd/tests/testdata/units/TEST-74-AUX-UTILS.mount.sh failed`

**Comment by yuwata** [comment]
> `[ 181.977183] TEST-74-AUX-UTILS.sh[428]: Subtest /usr/lib/systemd/tests/testdata/units/TEST-74-AUX-UTILS.mount.sh failed`

That's expected, as this PR does not fixes the issue.
Anyway, I will take a look.

File: test/units/TEST-74-AUX-UTILS.mount.sh [hunk 1]
```diff
@@ -43,6 +43,9 @@ mount "$LOOP" "$WORK_DIR/mnt"
 touch "$WORK_DIR/mnt/foo.bar"
 umount "$LOOP"
 (! mountpoint "$WORK_DIR/mnt")
+# Wait for the mount unit unloaded. Otherwise, creating transient unit below may be failed.
```
Comment (keszybz) [inline test/units/TEST-74-AUX-UTILS.mount.sh]: "Wait for the mount unit to be unloaded. Otherwise, creation of the transient unit may fail below."

**Review by keszybz — APPROVED** [review]
LGTM.

**Pull request merged**